crates/block-explorers/src/lib.rs

```diff
@@ -99,6 +115,16 @@ impl Client {
         Client::builder().with_api_key(api_key).chain(chain)?.build()
     }
 
+    /// Create a new client with the correct endpoint with the chain and chosen API v2 version
+    pub fn new_v2_from_env(chain: Chain) -> Result<Self> {
```

Comment:
I think we also want a way to set the version, like

```
fn with_version(self, v) -> Self
```

Reply:
This already exists:
```
    /// Configures the etherscan api version
    pub fn with_api_version(mut self, api_version: EtherscanApiVersion) -> Self {
        self.etherscan_api_version = api_version;
        self
    }
```